### ISSUE TITLE
Error when Gemfile is inside test directory

### ISSUE THREAD

**Issue by ankane** [ruby/setup-ruby#103]
Hi, thanks for this great action! I'm running into a pretty odd error when Gemfiles are inside the `test` directory (using Minitest), but not when they are in a different directory with the exact same code.

Here's a [minimal project](https://github.com/ankane/gemfile_repro) to reproduce and [the results](https://github.com/ankane/gemfile_repro/actions/runs/352875216) from GitHub Actions. There are 3 Gemfiles that all use the same code:

- `Gemfile` - works
- `gemfiles/Gemfile` - works
- `test/Gemfile` - errors with `File does not exist: pry` (<strike>but from what I can tell, none of the dependencies use pry</strike> it looks like `test/test_helper.rb` in `activerecord-import` tries to load pry)

Locally, all of the Gemfiles work, including the one in the test directory.

```sh
BUNDLE_GEMFILE=test/Gemfile bundle exec rake test
```

It looks like `$LOAD_PATH` changes on GitHub Actions but not locally when `BUNDLE_GEMFILE` changes, which may be related.

**Comment by eregon** [comment]
Using https://github.com/ruby/setup-ruby#working-directory might work better.

I think you should be able to reproduce this locally if you also do `bundle config --local path vendor/bundle` at the root of the repository, and follow the commands in the failed run.
Maybe it's an issue of Bundler not looking in the right place, when the Gemfile is not at the root.

Is there a specific reason you have a Gemfile not at the root of the repository? It seems more brittle in general, especially if you have both a Gemfile at the root and under `test/`.

**Comment by eregon** [comment]
One potential issue in `setup-ruby` is that the cache is reused, even though it's a different Gemfile file.
So probably we should include `BUNDLE_GEMFILE` in the cache key if it's set.

But anyway it's not the cause here, I can reproduce it locally:
```
clone
$ export BUNDLE_GEMFILE=test/Gemfile
$ bundle config --local path vendor/bundle
$ bundle check
$ bundle install --jobs 4 

$ bundle exec rake test
test Gemfile
test Gemfile
["/home/eregon/tmp/gemfile_repro/lib",
 "/home/eregon/tmp/gemfile_repro/test",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/rake-13.0.1/lib",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/activerecord-import-1.0.7/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/activemodel-6.0.3.4/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/zeitwerk-2.4.1/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/tzinfo-1.2.8/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/thread_safe-0.3.6/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/minitest-5.14.2/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/i18n-1.8.5/lib",
 "/home/eregon/tmp/gemfile_repro/test/vendor/bundle/ruby/2.7.0/gems/concurrent-ruby-1.1.7/lib/concurrent-ruby",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0/x86_64-linux",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/site_ruby",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/vendor_ruby/2.7.0",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/vendor_ruby/2.7.0/x86_64-linux",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/vendor_ruby",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/2.7.0",
 "/home/eregon/.rubies/ruby-2.7.2/lib/ruby/2.7.0/x86_64-linux"]

File does not exist: pry
```

So it must be an issue of Bundler, please report it there.

Something a bit interesting in this setup is that config gets saved to `test/.bundle/config` and gems to `test/vendor/bundle`, so everything seems relative to `BUNDLE_GEMFILE`.

**Comment by eregon** [comment]
I included the Gemfile path in the cache key in 693cf1794dd30e600179a37909578dece8011e20.
That won't solve the issue though.
The rest seems a Bundler bug and not an issue of setup-ruby.

**Comment by ankane** [comment]
Hey @eregon, thanks for looking into it (and all of your contributions to the Ruby community). Was able to reproduce locally, so will report to Bundler. Thanks!
